packages/sak-meny-henlegg/src/components/Brevmottakere.tsx

```diff
@@ -0,0 +1,67 @@
+import { required } from '@fpsak-frontend/utils';
+import { VerticalSpacer } from '@fpsak-frontend/shared-components';
+import React, { FunctionComponent, useEffect, useState } from 'react';
+import { ArbeidsgiverOpplysningerPerId, Personopplysninger } from '@k9-sak-web/types';
+import KlagePart from '@k9-sak-web/behandling-klage/src/types/klagePartTsType';
+import { SelectField } from '@fpsak-frontend/form/index';
+import { WrappedComponentProps } from 'react-intl';
+
+interface OwnProps {
+  hentMottakere: () => Promise<KlagePart[]>;
+  personopplysninger?: Personopplysninger;
+  arbeidsgiverOpplysninger?: ArbeidsgiverOpplysningerPerId;
+}
+
+function lagVisningsnavnForMottakere(
+  partId: string,
+  personopplysninger?: Personopplysninger,
+  arbeidsgiverOpplysningerPerId?: ArbeidsgiverOpplysningerPerId,
+): string {
+  if (
+    arbeidsgiverOpplysningerPerId &&
+    arbeidsgiverOpplysningerPerId[partId] &&
+    arbeidsgiverOpplysningerPerId[partId].navn
+  ) {
+    return `${arbeidsgiverOpplysningerPerId[partId].navn} (${partId})`;
+  }
+
+  if (personopplysninger && personopplysninger.aktoerId === partId) {
+    return `${personopplysninger.navn} (${personopplysninger.fnr || personopplysninger.nummer || partId})`;
+  }
+
+  return partId;
+}
+
+const Brevmottakere: FunctionComponent<OwnProps & WrappedComponentProps> = ({
+  hentMottakere,
+  personopplysninger,
+  arbeidsgiverOpplysninger,
+  intl,
+}) => {
+  const [mottakere, setMottakere] = useState<KlagePart[]>();
+
+  useEffect(() => {
```

Comment:
@magnus-bergman  endte opp med å bruke `useEffect()` og `useState()` for å få det til å funke. Er det noe bedre måte?

Reply:
useState og useEffect er bare fint å bruke, men spørsmålet er om vi trenger å håndtere feilede request på samme sted eller om de catches globalt i useApi hooken. Hva skjer feks om requesten feiler eller det ikke finnes noen klageparter, kan man fortsatt henlegge da?

Reply:
@magnus-bergman  hvis det feiler så bør henleggelse feile. Da blir valgtPart = null ihvertfall, men hvordan feilmelding vises er jeg usikker på

Reply:
hvis klageparter array er tom så vises ikke det dropdown og det er riktig for da skal man ikke velge mottaker